### PR TITLE
More unit tests

### PR DESCRIPTION
**Pull request by RoyEJohnson**

### CODE REVIEW

**Comment by philschatz** [comment]
o no, why did coverage decrease?

**Comment by RoyEJohnson** [comment]
@philschatz Don't know. It also decreased by a fraction in the "Set Salesforce home" PR. In that case, it's probably because I added a line to an imported-but-untested area. In this case, maybe the new tests import too much untested code.

**Comment by RoyEJohnson** [comment]
I also can't get it to pass tests on Travis that I can get it to pass locally.

File: package.json [hunk 1]
```diff
@@ -73,6 +73,7 @@
       "settings": "<rootDir>/dist/settings",
       "fetch": "<rootDir>/jspm_packages/npm/whatwg-fetch@1.1.1/fetch.js"
     },
+    "coveragePathIgnorePatterns": ["/analytics", "/recordo", "/whatwg-fetch", "/router"],
```
Comment (philschatz) [inline package.json]: Could this just be the `jspm_modules` directory?

File: src/app/pages/subjects/subjects.js [hunk 2]
```diff
@@ -31,12 +31,10 @@ export default class Subjects extends CMSPageController {
 
         shell.showLoader();
 
-        if (!$.isNode()) {
-            router.replaceState({
-                filter: this.categoryFromPath(),
-                path: pagePath
-            });
-        }
+        router.replaceState({
```
Comment (philschatz) [inline src/app/pages/subjects/subjects.js]: ooh, no more `$.isNode()`? How'd that happen?

File: src/app/pages/subjects/subjects.js [hunk 3]
```diff
@@ -55,15 +53,13 @@ export default class Subjects extends CMSPageController {
 
         this.bookViewer.filterSubjects(category);
 
-        if (!$.isNode()) {
-            const yTarget = history.state.y;
+        const yTarget = history.state.y;
```
Comment (philschatz) [inline src/app/pages/subjects/subjects.js]: hmm, could/should this be `window.history` to clarify that this isn't a local variable?

Comment (RoyEJohnson) [inline src/app/pages/subjects/subjects.js]: Could be, but there are many uses of history throughout the code base already.

File: package.json (hunk 1 above)
Comment (RoyEJohnson) [inline package.json]: Yes, apparently. :)

File: test/setupFile.js [hunk 4]
```diff
@@ -3,6 +3,7 @@ import 'babel-polyfill';
 jest.mock('~/components/shell/shell');
 
 global.fetch = require('jest-fetch-mock');
+global.regeneratorRuntime = regeneratorRuntime;
```
Comment (philschatz) [inline test/setupFile.js]: OIC, cool!

And it looks like the line above could be switched to `require('fetch-vcr')` and you can record the HTTP responses from Django 😄

Comment (RoyEJohnson) [inline test/setupFile.js]: That's an outdated change, but there is a different fetch package that could be swapped out.

**Pull request merged**